src/EditorFeatures/Core/Implementation/CodeCleanup/CodeCleanupService.cs

```diff
@@ -0,0 +1,165 @@
+                IDEDiagnosticIds.UseExpressionBodyForMethodsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForConversionOperatorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForOperatorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForPropertiesDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixInlineVariableDeclarations,
+                new[] { IDEDiagnosticIds.InlineDeclarationDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.RemoveUnusedVariables,
+                new[] { "CS0168", "CS0219" });
+            dictionary.Add(FeatureOnOffOptions.FixObjectCollectionInitialization,
+                new[] { IDEDiagnosticIds.UseObjectInitializerDiagnosticId, IDEDiagnosticIds.UseCollectionInitializerDiagnosticId });
+            //dictionary.Add(FeatureOnOffOptions.FixLanguageFeatures,
+            //    new[] { IDEDiagnosticIds. });
+            //IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInDeclarationsDiagnosticId,
+            //    IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInMemberAccessDiagnosticId,
+            //    IDEDiagnosticIds.InlineAsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeWithoutNameCheckDiagnosticsId,
+
+            return dictionary.ToImmutableDictionary();
+        }
+
+        public Task<IEnumerable<TextChange>> GetChangesForCleanupDocument(Document document, CancellationToken cancellationToken)
+        {
+            var oldDocument = document;
+            document = RemoveSortUsings(document, cancellationToken);
+            document = ApplyCodeFixes(document, cancellationToken);
+
+            return document.GetTextChangesAsync(oldDocument, cancellationToken);
+        }
+        private Document RemoveSortUsings(Document document, CancellationToken cancellationToken)
+        {
+            // remove and sort usings
+            if (document.Project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.RemoveUnusedUsings, LanguageNames.CSharp))
+            {
+                var removeUsingsService = document.GetLanguageService<IRemoveUnnecessaryImportsService>();
+                if (removeUsingsService != null)
+                {
+                    document = removeUsingsService.RemoveUnnecessaryImportsAsync(document, cancellationToken).WaitAndGetResult(cancellationToken);
+                }
+            }
+
+            // sort usings
+            if (document.Project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.SortUsings, LanguageNames.CSharp))
+            {
+                document = OrganizeImportsService.OrganizeImportsAsync(document, cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
this will bock the thread we're on.  Is there a reason all this code isn't just async from end-to-end?

Reply:
fixed